chain/src/sumtree.rs

```diff
@@ -252,7 +287,8 @@ impl<'a> Extension<'a> {
 		}
 
 		for out in &b.outputs {
-			if let Ok(_) = self.commit_index.get_output_pos(&out.commitment()) {
+			// if let Ok(_) = self.commit_index.get_output_pos(&out.commitment()) {
+			if let Ok(true) = self.is_duplicate_output(&out.commitment()) {
```

Comment:
is_duplicate_commit checks both the index and the rewound pmmr - checking to see if the commit at that pos is what we expect.